_tools/libddwaf-updater/update.go

```diff
@@ -0,0 +1,284 @@
+		embedDir := path.Join(libDir, fmt.Sprintf("%s-%s", tgt.os, tgt.arch))
+		created := false
+		if _, err = os.Stat(embedDir); errors.Is(err, os.ErrNotExist) {
+			if err = os.MkdirAll(embedDir, 0755); err != nil {
+				panic(err)
+			}
+			created = true
+		}
+		if created || force {
+			createEmbedSource(tgt)
+		}
+		go handleTarget(&wg, version, tgt, embedDir, assets)
+	}
+
+	wg.Wait()
+
+	file, err := os.OpenFile(versionFile, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0644)
+	if err != nil {
+		panic(err)
+	}
+	written := 0
+	for written < len(version) {
+		wrote, err := file.WriteString(version[written:])
+		if err != nil {
+			panic(err)
+		}
+		written += wrote
+	}
+
+	fmt.Println("All done! Don't forget to check in changes to include/ and internal/vendor/, check the libddwaf upgrade guide to update bindings!")
+}
+
+func createEmbedSource(tgt target) {
+	ext := "so"
+	if tgt.os == "darwin" {
+		ext = "dylib"
+	}
+
+	gosource := strings.Join(
+		[]string{
+			"// Unless explicitly stated otherwise all files in this repository are licensed",
+			"// under the Apache License Version 2.0.",
+			"// This product includes software developed at Datadog (https://www.datadoghq.com/).",
+			"// Copyright 2016-present Datadog, Inc.",
+			"",
+			fmt.Sprintf("//go:build %s && %s && !go1.22", tgt.os, tgt.arch),
+			"package vendor",
+			"",
+			`import _ "embed" // Needed for go:embed`,
+			"",
+			fmt.Sprintf("//go:embed %s-%s/libddwaf.%s", tgt.os, tgt.arch, ext),
+			"var libddwaf []byte",
+			"",
+			fmt.Sprintf(`const embedNamePattern = "libddwaf-*.%s"`, ext),
+			"", // Trailing new line...
+		},
+		"\n",
+	)
+	if err := os.WriteFile(path.Join(libDir, fmt.Sprintf("vendor_%s_%s.go", tgt.os, tgt.arch)), []byte(gosource), 0644); err != nil {
+		panic(err)
+	}
+}
+
+func handleTarget(wg *sync.WaitGroup, version string, tgt target, embedDir string, assets map[string]*github.ReleaseAsset) {
+	defer wg.Done()
+
+	tarName := fmt.Sprintf("libddwaf-%s-%s.tar.gz", version, tgt.assetLabel)
+	shaName := fmt.Sprintf("%s.sha256", tarName)
+
+	tarAsset, found := assets[tarName]
+	if !found {
+		panic(fmt.Errorf("could not find tarball named %s", tarName))
+	}
+	shaAsset, found := assets[shaName]
+	if !found {
+		panic(fmt.Errorf("could not find sha256 named %s", shaName))
+	}
+
+	tarUrl := *tarAsset.BrowserDownloadURL
+	shaUrl := *shaAsset.BrowserDownloadURL
+
+	tmpdir, err := os.MkdirTemp("", "libddwaf-*")
+	if err != nil {
+		panic(err)
+	}
+	defer os.RemoveAll(tmpdir)
+
+	if _, err := script.Get(tarUrl).WriteFile(path.Join(tmpdir, tarName)); err != nil {
+		panic(err)
+	}
+	sha, err := script.Get(shaUrl).String()
+	if err != nil {
+		panic(err)
+	}
+	sum, err := script.File(path.Join(tmpdir, tarName)).SHA256Sum()
+	if err != nil {
+		panic(err)
+	}
+	// To match the shasum format...
+	sum = fmt.Sprintf("%s  %s\n", sum, tarName)
+	if sum != sha {
+		panic(fmt.Errorf("checksum mismatch on %s:\nExpected %s\nActual   %s", tarUrl, sha, sum))
```

Comment:
```suggestion
		panic(fmt.Errorf("checksum mismatch on %s:\nExpected %s\nActual  %s", tarUrl, sha, sum))
```

Reply:
Copied from my (original) version, aligned... shows that no change is necessary:

```
Expected %s
Actual   %s
```